group/assignment5/AlphabetFromDictionary.cpp

```diff
@@ -1,120 +1,175 @@
 #include "AlphabetFromDictionary.hpp"
 
 AlphabetFromDictionary::AlphabetFromDictionary(const vector<string> &dictionary) : dictionary(dictionary) {
-    this->setDictionary(dictionary);
+	this->setDictionary(dictionary);
 }
 
 void AlphabetFromDictionary::setDictionary(const vector<string> &dictionary) {
-    this->dictionary = dictionary;
-    this->initializeClassVars();
-    this->findAlphabet();
+	this->dictionary = dictionary;
+	this->initializeClassVars();
+	this->findAlphabet();
 }
 
 vector<char> AlphabetFromDictionary::getAlphabet() {
-    return this->orderedAlphabet;
+	return this->orderedAlphabet;
 }
 
 bool AlphabetFromDictionary::isDictionaryConsistent() {
-    return !this->isGraphCyclic;
+	return !this->isGraphCyclic;
 }
 
 void AlphabetFromDictionary::initializeClassVars() {
-    this->findAlphabetCharacters();
-    this->isGraphCyclic = false;
-    this->orderedAlphabet.clear();
-    this->charPriorityGraph.clear();
-    for (auto c : this->alphabetCharacters) {
-        this->charPriorityGraph[c] = vector<bool>(NUM_OF_NATIVE_CHARS, false);
-    }
+	this->findAlphabetCharacters();
+	this->isGraphCyclic = false;
+	this->orderedAlphabet.clear();
+	this->charPriorityGraph.clear();
+	for (auto c : this->alphabetCharacters) {
+		this->charPriorityGraph[c] = vector<bool>(NUM_OF_NATIVE_CHARS, false);
+	}
+	this->inDegree.resize(NUM_OF_NATIVE_CHARS, 0);

```

Comment:
When you call resize with a value (0), it adds the number of needed elements, and only the added elements receive the specified value (0).
That means that when we set a new dictionary and the size of inDegree is already 256, this line does nothing and the vector is not initialized, and it accumulates the values from previous runs.
One way to fix that is to replace this line with "this->inDegree = vector<int>(NUM_OF_NATIVE_CHARS, 0);".
This way all the elements in inDegree are 0 after initializeClassVars() is called.